core/src/main/java/org/apache/accumulo/core/clientImpl/ClientContext.java

```diff
@@ -164,13 +176,51 @@ public ClientContext(SingletonReservation reservation, ClientInfo info,
     this.singletonReservation = Objects.requireNonNull(reservation);
     this.tableops = new TableOperationsImpl(this);
     this.namespaceops = new NamespaceOperationsImpl(this, tableops);
+    if (ueh == Threads.UEH) {
+      this.ueh = ueh;
+      clientThreadPools = ThreadPools.getServerThreadPools();
+    } else {
+      // Provide a default UEH that just logs the error
+      if (ueh == null) {
+        this.ueh = (t, e) -> {
+          log.error("Caught an Exception in client background thread: {}. Thread is dead.", t, e);
+        };
+      } else {
+        this.ueh = ueh;
+      }
+      clientThreadPools = ThreadPools.getClientThreadPools(ueh);
+    }
   }
 
   public Ample getAmple() {
     ensureOpen();
     return new AmpleImpl(this);
   }
 
+  public synchronized Future<List<KeyValue>>
+      submitScannerReadAheadTask(Callable<List<KeyValue>> c) {
+    if (scannerReadaheadPool == null) {
+      scannerReadaheadPool = clientThreadPools.createThreadPool(0, Integer.MAX_VALUE, 3L, SECONDS,
+          "Accumulo scanner read ahead thread", new SynchronousQueue<>(), true);
+    }
```

Comment:
Could consider creating this in the constructor, then would not need to synchronize the method.

Reply:
@keith-turner I believe that's what he did in a previous iteration, but I suggested he lazily instantiate, to follow the convention established with the thriftTransportPool which avoids allocating resources which might not be used by the context.

Reply:
That's correct.

Reply:
As long as closed() is synchronized, that addresses the main concern I had.